src/app/main.rs

```diff
@@ -839,33 +824,30 @@ where
                 .layer(transport_metrics.accept("inbound"))
                 .layer(keepalive::accept::layer(config.inbound_accept_keepalive));
 
-            serve(
+            spawn_server(
                 "in",
                 inbound_listener,
                 accept,
                 connect,
                 source_stack,
                 config.h2_settings,
                 drain_rx.clone(),
-            )
-            .map_err(|e| error!("inbound proxy background task failed: {}", e))
+            );
         };
-        task::spawn(inbound);
     }
 }
 
 type Error = Box<dyn std::error::Error + Send + Sync>;
 
-fn serve<A, T, C, R, B, G>(
+fn spawn_server<A, T, C, R, B, G>(
```

Comment:
What do you think about just getting rid of this function, and just constructing a `proxy::Server` and calling `proxy::spawn` with that server in  inbound/outbound? This doesn't seem to be doing much of anything anymore, and just creates another place where we have to have a big list of `where` bounds. It would be nice to reduce the number of places where we have to maintain a giant list of bounds, especially if those constraints are enforced elsewhere as well.

Reply:
it's simplified substantially in the followup

Reply:
okay, cool; carry on then!